### PR TITLE
Refactor organise admin-ui code

### PR DESCRIPTION
**Pull request by joonatank**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Reorganise `admin-ui` so pages are under `spa` not `components`
- Remove unnecessary `edit`, `/recurring-reservations`, and `/premises-and-settings/`  from urls.
- Refactor: use url builders for all internal links.
- Remove unnecessary spaces and resources listing pages that were not linked from anywhere.
- Remove copy of a HDS Table component (use HDS instead).

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Manual testing: check most common functionality. Links and tables (sorting) refactored. `admin-ui` only

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- [TILA-2796](https://helsinkisolutionoffice.atlassian.net/browse/TILA-2796)

[TILA-2796]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-2796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/admin-ui/src/spa/my-units/[id]/CreateReservationModal.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ import {
   type ReservationFormMeta,
 } from "@/schemas";
 import { breakpoints } from "common/src/common/style";
-import { useCheckCollisions } from "@/component/reservations/requested/hooks";
+import { useCheckCollisions } from "@//hooks";
```
Comment (joonatank) [inline apps/admin-ui/src/spa/my-units/[id]/CreateReservationModal.tsx:24]: `//`

File: apps/admin-ui/src/spa/unit/[id]/ReservationUnitCard.tsx [hunk 2]
```diff
@@ -113,8 +114,7 @@ export function ReservationUnitCard({
 
   const hasPurposes = (reservationUnit?.purposes?.length || 0) > 0;
 
-  // TODO use urlBuilder
-  const link = `/unit/${unitId}/reservationUnit/edit/${reservationUnit.pk}`;
+  const link = getReservationUnitUrl(reservationUnit.pk, unitId);
```
Comment (joonatank) [inline apps/admin-ui/src/spa/unit/[id]/ReservationUnitCard.tsx:117]: `?` is it edit or the reservationUnit page? what tells the difference in url builder?

File: apps/admin-ui/src/spa/unit/[id]/index.tsx [hunk 3]
```diff
@@ -188,7 +189,7 @@ function Unit(): JSX.Element {
           variant="supplementary"
           iconLeft={<IconPlusCircleFill />}
           onClick={() => {
-            history(`/unit/${unitPk}/reservationUnit/edit/`);
+            history(getReservationUnitUrl(undefined, unitPk));
```
Comment (joonatank) [inline apps/admin-ui/src/spa/unit/[id]/index.tsx:192]: ?

File: apps/admin-ui/tsconfig.json [hunk 4]
```diff
@@ -4,7 +4,6 @@
     "plugins": [{ "name": "next" }],
     "paths": {
       // old version
```
Comment (joonatank) [inline apps/admin-ui/tsconfig.json:6]: remove comment

File: apps/admin-ui/src/spa/unit/[id]/ReservationUnitCard.tsx (hunk 2 above)
Comment (joonatank) [reply apps/admin-ui/src/spa/unit/[id]/ReservationUnitCard.tsx:117]: yeah it's correct, naming is maybe questionable

**Review by vincit-matu — APPROVED** [review]

**Pull request merged**